crates/core/src/factories/core_space.rs

```diff
@@ -5,6 +5,55 @@ use std::sync::{Arc, Mutex, Weak};
 
 mod protocol;
 
+/// CoreSpace configuration types.
+pub mod config {
+    /// Configuration parameters for [CoreSpaceFactory](super::CoreSpaceFactory).
+    #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct CoreSpaceConfig {
+        /// The interval in millis at which we check for about to expire
+        /// local agent infos.
+        /// Default: 60s.
+        pub re_sign_freq_ms: u32,
+
+        /// The time in millis before an agent info expires after which we will
+        /// re sign them.
+        /// Default: 5m.
+        pub re_sign_expire_time_ms: u32,
```

Comment:
What is this for? If an agent expires within this period of time and the check task runs, they'll send their info to the bootstrap again. If an agent expires after this period, they won't send the info. Why?

Reply:
No, it's the other way around. This is making sure we wait *at least* until five minutes before expiry before re-signing a new one.

Reply:
Right yes, I wrote it wrongly. What's the purpose of this? Or configuring this?

Reply:
Honestly? Because every time I just hard-code something people ask it to be in the config before approving.

Reply:
Sorryyyyyy :) I think because it interacts with the check interval and the agent info expiry, it's probably sensible to have this in configuration but can see why it would be find to hard-code it too

Reply:
Those damn people!